File: app/src/main/java/net/vornao/ddns/dome/devices/EnergyMonitor.java
package net.vornao.ddns.dome.devices;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import net.vornao.ddns.dome.callbacks.EnergyMonitorUpdatedCallback;
import net.vornao.ddns.dome.handler.DeviceHandler;

public class EnergyMonitor extends Device{

    private float power;
    private float energy;


    @Nullable
    private EnergyMonitorUpdatedCallback emonCallback = null;
    private String dataUrl;

    public EnergyMonitor(int id, String type, DeviceHandler handler, String home_id, String name) {
        super(id, type, handler, home_id, name);
    }

    public float getPower() {
        return power;
    }

    public void setPower(float power) {
        this.power = power;
        if(emonCallback != null) emonCallback.onUpdate();
    }

    public float getEnergy() {
        return energy;
    }

    public void setEnergy(float energy) {
        this.energy = energy;
    }

    public String getDataUrl() {
        return dataUrl;
    }

    public void setDataUrl(String dataUrl) {
        this.dataUrl = dataUrl;
    }

    @Nullable
    public EnergyMonitorUpdatedCallback getEmonCallback() {
        return emonCallback;
    }

    public void setEmonCallback(EnergyMonitorUpdatedCallback emonCallback) {
        this.emonCallback = emonCallback;
    }

    @NonNull
    @Override
    public String toString() {
        return "EnergyMonitor{" +
                "power=" + power +
                ", energy=" + energy +
                ", dataUrl='" + dataUrl + '\'' +
                '}';
    }
}

File: app/src/main/java/net/vornao/ddns/dome/devices/EnergyMonitor.kt
package net.vornao.ddns.dome.devices
import net.vornao.ddns.dome.handler.DeviceHandler

class EnergyMonitor(id: Int, type: String, handler: DeviceHandler, home_id: String, name: String) : Device(id, type, handler, home_id, name) {
    var power = 0f
     set(value) {
         field = value
         emonCallback?.invoke()
     }


    var energy = 0f
    var emonCallback: (()->Unit)? = null
    var dataUrl: String? = null

    override fun toString(): String {
        return "EnergyMonitor{" +
                "power=" + power +
                ", energy=" + energy +
                ", dataUrl='" + dataUrl + '\'' +
                '}'
    }
}

